packages/augur-sdk/src/state/getter/Markets.ts

```diff
@@ -1365,3 +1375,129 @@ async function getMarketsSearchResults(
   }
   return Augur.syncableFlexSearch.where(whereObj);
 }
+
+/**
+ * Sets the `lastTradedTimestamp` property for all markets in `marketsResults`.
+ *
+ * @param {DB} db Database object to use for getting `lastTradedTimestamp` info
+ * @param {any[]} marketsResults Array of market objects to add `lastTradedTimestamp` to
+ */
+async function setLastTradedTimestamp(db: DB, marketsResults: any[]): Promise<Array<{}>>  {
+  // Create market ID => marketsResults index mapping
+  const keyedMarkets = {};
+  for (let i = 0; i < marketsResults.length; i++) {
+    keyedMarkets[marketsResults[i].market] = i;
+  }
+
+  const orderFilledLogs = await db.findOrderFilledLogs({
+    selector: { market: { $in: marketsResults.map(marketsResult => marketsResult.market) } },
+    fields: ['market', 'timestamp'],
+  });
+
+  // Assign lastTradedTimestamp value to each market in marketsResults
+  for (let i = 0; i < orderFilledLogs.length; i++) {
+    const currentTimestamp = parseInt(orderFilledLogs[i].timestamp, 16);
+    if (
+      !marketsResults[keyedMarkets[orderFilledLogs[i].market]][GetMarketsSortBy.lastTradedTimestamp] ||
+      currentTimestamp > marketsResults[keyedMarkets[orderFilledLogs[i].market]][GetMarketsSortBy.lastTradedTimestamp]
+    ) {
+      marketsResults[keyedMarkets[orderFilledLogs[i].market]][GetMarketsSortBy.lastTradedTimestamp] = currentTimestamp;
+    }
+  }
+
+  // For any markets with no trading, set the lastTradedTimestamp to 0
+  for (let i = 0; i < marketsResults.length; i++) {
+    if (!marketsResults[i][GetMarketsSortBy.lastTradedTimestamp]) {
+      marketsResults[i][GetMarketsSortBy.lastTradedTimestamp] = 0;
+    }
+  }
+  return marketsResults;
+}
+
+/**
+ * Sets the `liquidity` property for the last 24 hours of liquidity for all markets in `marketsResults`.
+ *
+ * @param {DB} db Database object to use for getting `liquidity` info
+ * @param {Array<Object>} marketsResults Array of market objects to add `liquidity` to
+ */
+async function setLiquidity(db: DB, marketsResults: any[]): Promise<Array<{}>>  {
```

Comment:
way `any` here for `marketsResults` and not typed?
Also with return value way empty object `{}`?

Reply:
I agree we should ideally be using an actual type here, but for some reason, FlexSearch is returning a different type than the expected Array<SearchResults<MarketFields>>, which in turn means I have to define the type as any[] as a workaround. This will take a fair amount of time to figure out why this is happening, so I've created a TODO item to come back to this.